### PR TITLE
"[backend_49] Email Service"

### PR DESCRIPTION
**Pull request by sanketshah1997**
## Description

<Please include a brief description of the code changes>

## Issue

Fixes #<issue_number>

## Checklist
- [x] I have commented my code in hard-to-understand areas
- [x] I have created relevant Unit Tests for the feature I implemented
- [x] I have pulled and merged the latest code and resolved all merge conflicts
- [x] I have added someone to review this PR, and informed the team on Slack

### CODE REVIEW

**Review by Nishit-Gopani — APPROVED** [review]
Looks good :)

File: src/main/java/com/jjcsa/service/EmailSenderService.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class EmailSenderService {
```
Comment (shah-tejas) [inline src/main/java/com/jjcsa/service/EmailSenderService.java:17]: This class looks pretty good. In future, we should add unit tests for this class, mocking the external services.

File: src/main/java/com/jjcsa/service/EmailTemplateService.java [hunk 2]
```diff
@@ -1,40 +1,32 @@
...
-import java.io.Serializable;
 import java.util.HashMap;
+import java.util.Locale;
 import java.util.Map;
 
 @Service
 @Slf4j
-@NoArgsConstructor
-@AllArgsConstructor
+@Data
```
Comment (shah-tejas) [inline src/main/java/com/jjcsa/service/EmailTemplateService.java:19]: This will create getters and setters for the properties. is it needed?

**Review by shah-tejas — APPROVED** [review]
Looks pretty good. Some small suggestions (no important)

Also, I was thinking to support local development we can use [localstack](https://hub.docker.com/r/localstack/localstack) to mock AWS services. It would simply be adding another container to 
https://github.com/JJCSA/backend/blob/6513c0464fec7f56336fd838d2787fb98ef55088/postgres/docker-compose.yml#L3

File: src/main/java/com/jjcsa/service/EmailSenderService.java (hunk 1 above)
Comment (sanketshah1997) [reply src/main/java/com/jjcsa/service/EmailSenderService.java:17]: Agree. Need to have a habit of writing unit tests with code!

File: src/main/java/com/jjcsa/service/EmailTemplateService.java (hunk 2 above)
Comment (Nishit-Gopani) [reply src/main/java/com/jjcsa/service/EmailTemplateService.java:19]: Sanket can you update this and update the PR. I will merged once it is done. Thanks

**Pull request merged**